### PR TITLE
Test otel metrics with fault tolerance

### PR DESCRIPTION
**Pull request by Yas-Franco**
Test otel metrics with fault tolerance

### CODE REVIEW

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/bnd.bnd [hunk 1]
```diff
@@ -0,0 +1,55 @@
+#*******************************************************************************
+# Copyright (c) 2017, 2020 IBM Corporation and others.
```
Comment (Yas-Franco) [inline dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/bnd.bnd]: Copyright date should be 2024

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/bnd.bnd [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+-buildpath: \
+	../build.sharedResources/lib/junit/old/junit.jar;version=file, \
+    org.jboss.shrinkwrap:shrinkwrap-impl-base;version=1.2.3,\
+    org.jboss.shrinkwrap:shrinkwrap-spi;version=1.2.3,\
+	com.ibm.websphere.javaee.ejb.3.2;version=latest,\
+	com.ibm.websphere.javaee.servlet.3.1;version=latest,\
+	io.openliberty.jakarta.servlet.6.0;version=latest,\
+	com.ibm.websphere.org.eclipse.microprofile.faulttolerance.1.1;version=latest,\
+	com.ibm.websphere.org.eclipse.microprofile.config.1.3;version=latest,\
+	com.ibm.websphere.org.eclipse.microprofile.metrics.1.0;version=latest,\
```
Comment (Yas-Franco) [inline dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/bnd.bnd]: The indentation here is inconsistent

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/build.gradle [hunk 3]
```diff
@@ -0,0 +1,20 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2021 IBM Corporation and others.
```
Comment (Yas-Franco) [inline dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/build.gradle]: Copyright date should be 2024

File: dev/com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0/bnd.bnd [hunk 4]
```diff
@@ -0,0 +1,59 @@
+#*******************************************************************************
+# Copyright (c) 2018 IBM Corporation and others.
```
Comment (Yas-Franco) [inline dev/com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0/bnd.bnd]: 2024

File: dev/com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0/bnd.bnd [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/bundle.props
+bVersion=1.0
+
+openTelemetryVersion=1.35.0
+
+Bundle-Name: com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0
```
Comment (Yas-Franco) [inline dev/com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0/bnd.bnd]: This bundle should probably be called: 

io.openliberty.microprofile.faulttolerance.4.0.telemetry.metrics.2.0

File: dev/com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0/src/com/ibm/ws/microprofile/faulttolerance/telemetry/metrics_20/MetricRecorderImpl.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
```
Comment (Yas-Franco) [inline dev/com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0/src/com/ibm/ws/microprofile/faulttolerance/telemetry/metrics_20/MetricRecorderImpl.java]: 2024

File: dev/com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0/bnd.bnd [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        com.ibm.ws.logging.core;version=latest, \
+        com.ibm.ws.kernel.service;version=latest,\
+        org.eclipse.osgi;version=latest, \
+        com.ibm.websphere.org.osgi.core;version=latest, \
+        com.ibm.websphere.org.osgi.service.component;version=latest, \
+        com.ibm.wsspi.org.osgi.service.component.annotations;version=latest, \
+        com.ibm.ws.microprofile.faulttolerance;version=latest, \
+        com.ibm.ws.microprofile.faulttolerance.spi;version=latest, \
+        com.ibm.websphere.org.eclipse.microprofile.metrics.2.0;version=latest, \
+        com.ibm.websphere.org.eclipse.microprofile.config.1.3;version=latest,\
```
Comment (Yas-Franco) [inline dev/com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0/bnd.bnd]: Indentations are inconsistent

File: dev/com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0/src/com/ibm/ws/microprofile/faulttolerance/telemetry/metrics_20/MetricRecorderImpl.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+import com.ibm.ws.microprofile.faulttolerance.spi.BulkheadPolicy;
+import com.ibm.ws.microprofile.faulttolerance.spi.CircuitBreakerPolicy;
+import com.ibm.ws.microprofile.faulttolerance.spi.FallbackPolicy;
+import com.ibm.ws.microprofile.faulttolerance.spi.MetricRecorderProvider.AsyncType;
+import com.ibm.ws.microprofile.faulttolerance.spi.RetryPolicy;
+import com.ibm.ws.microprofile.faulttolerance.spi.TimeoutPolicy;
+
+/**
+ * Records metrics using MP Metrics API 2.0
```
Comment (Yas-Franco) [inline dev/com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0/src/com/ibm/ws/microprofile/faulttolerance/telemetry/metrics_20/MetricRecorderImpl.java]: MpTelemetry 2.0 metrics (Not mp metrics)

File: dev/com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0/src/com/ibm/ws/microprofile/faulttolerance/telemetry/metrics_20/MetricRecorderProviderImpl.java [hunk 9]
```diff
@@ -0,0 +1,125 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
```
Comment (Yas-Franco) [inline dev/com.ibm.ws.microprofile.faulttolerance.telemetry.metrics.2.0/src/com/ibm/ws/microprofile/faulttolerance/telemetry/metrics_20/MetricRecorderProviderImpl.java]: 2024

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/publish/servers/CDIFaultToleranceMetricsRemoval/server.xml [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+	    <feature>componenttest-1.0</feature>
+		<feature>osgiconsole-1.0</feature>
+		<feature>localConnector-1.0</feature>
+		<feature>cdi-4.0</feature>
+		<feature>servlet-6.0</feature>
+		<feature>mpFaultTolerance-4.0</feature>
+		<feature>mpTelemetry-2.0</feature>
+	</featureManager>
+	
+	<mpMetrics authentication="false"/>
```
Comment (Yas-Franco) [inline dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/publish/servers/CDIFaultToleranceMetricsRemoval/server.xml]: Is this still needed?

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/publish/servers/CDIFaultToleranceMetricsRemoval/server.xml [hunk 11]
```diff
@@ -0,0 +1,29 @@
+<!--
+    Copyright (c) 2017, 2020 IBM Corporation and others.
```
Comment (Yas-Franco) [inline dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/publish/servers/CDIFaultToleranceMetricsRemoval/server.xml]: 2024

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/publish/servers/CDIFaultToleranceCircuitBreakerMetricsRemoval/server.xml [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+		<feature>mpFaultTolerance-4.0</feature>
+		<feature>mpTelemetry-2.0</feature>
+	</featureManager>
+	
+	<application type="war" location="ftCircuitBreakerMetrics.war">
+		<classloader apiTypeVisibility="+third-party" />
+	</application>
+
+	
+	<mpMetrics authentication="false"/>
```
Comment (Yas-Franco) [inline dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/publish/servers/CDIFaultToleranceCircuitBreakerMetricsRemoval/server.xml]: Is this still needed?

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/fat/src/com/ibm/websphere/microprofile/faulttolerance/metrics/fat/suite/FATSuite.java [hunk 13]
```diff
@@ -0,0 +1,29 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2021 IBM Corporation and others.
```
Comment (Yas-Franco) [inline dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/fat/src/com/ibm/websphere/microprofile/faulttolerance/metrics/fat/suite/FATSuite.java]: 2024

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/fat/src/com/ibm/websphere/microprofile/faulttolerance/metrics/fat/tests/CDIFallbackTest.java [hunk 14]
```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2021 IBM Corporation and others.
```
Comment (Yas-Franco) [inline dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/fat/src/com/ibm/websphere/microprofile/faulttolerance/metrics/fat/tests/CDIFallbackTest.java]: 2024

File: dev/com.ibm.ws.microprofile.faulttolerance.telemetry/bnd.bnd [hunk 15]
```diff
@@ -0,0 +1,52 @@
+#*******************************************************************************
+# Copyright (c) 2018 IBM Corporation and others.
```
Comment (Yas-Franco) [inline dev/com.ibm.ws.microprofile.faulttolerance.telemetry/bnd.bnd]: 2024

File: dev/com.ibm.ws.microprofile.faulttolerance.telemetry/bnd.bnd [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/bundle.props
+bVersion=1.0
+
+openTelemetryVersion=1.35.0
+
+Bundle-Name: com.ibm.ws.microprofile.faulttolerance.telemetry
```
Comment (Yas-Franco) [inline dev/com.ibm.ws.microprofile.faulttolerance.telemetry/bnd.bnd]: Should probably be 

io.openliberty.microprofile.faulttolerance.telemetry

File: dev/com.ibm.ws.microprofile.faulttolerance.telemetry/src/com/ibm/ws/microprofile/faulttolerance/telemetry/metrics/integration/package-info.java [hunk 17]
```diff
@@ -0,0 +1,20 @@
+/*******************************************************************************
+ * Copyright (c) 2018 IBM Corporation and others.
```
Comment (Yas-Franco) [inline dev/com.ibm.ws.microprofile.faulttolerance.telemetry/src/com/ibm/ws/microprofile/faulttolerance/telemetry/metrics/integration/package-info.java]: 2024

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/test-applications/faulttolerancemetrics.jar/src/com/ibm/websphere/microprofile/faulttolerance/metrics/utils/CustomException.java [hunk 18]
```diff
@@ -0,0 +1,30 @@
+/*******************************************************************************
+ * Copyright (c) 2017 IBM Corporation and others.
```
Comment (Yas-Franco) [inline dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/test-applications/faulttolerancemetrics.jar/src/com/ibm/websphere/microprofile/faulttolerance/metrics/utils/CustomException.java]: 2024

Comment (benjamin-confino) [reply dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_telemetry_metrics/test-applications/faulttolerancemetrics.jar/src/com/ibm/websphere/microprofile/faulttolerance/metrics/utils/CustomException.java]: I don't think we need to change this one, its copying a file unchanged

**Comment by benjamin-confino** [comment]
#build

**Pull request closed without merging**